src/blockdata/transaction.rs

```diff
@@ -673,7 +689,9 @@ impl SigHashType {
          }
      }
 
-     /// Reads a 4-byte uint32 as a sighash type
+     /// Reads a 4-byte uint32 as a sighash type.
+     /// Note: this replicates consensus behaviour, for current standardness rules correctness
+     /// you probably want [from_u32_standard].
```

Comment:
If this is critical for certain applications, maybe this should be deprecated, and `from_u32_standard`, `from_u32_nonstandard` used instead?

Reply:
Yeah, marking it as deprecated and introducing the new method to eventually remove the old one sounds like a good plan.

Reply:
I think it's good to still have one that matches Consensus :tm: behaviour ? But yeah would be great to shift them `from_u32 -> Result<u32>`, `from_u32_consensus -> u32`, but imho too much bothering

Reply:
I think the idea was to get rid of `from_u32` entirely eventually, because it's ambiguous.

Reply:
Yeap. `from_u32` is such a standard name that no one is going to pay attention and just use it, potentially introducing bugs. If there are `from_u32_consensus` and `from_u32_standard` then the user gets to scratch their head a bit looking at the autocompletion suggesting two variants, and is going to read the documentation and hopefully pick the right one.

First deprecated `from_u32` to make the transition less painful, then remove it completely.

Reply:
Right, added a commit to introduce `from_u32_consensus` and deprecate `from_u32`